projects/packages/blocks/tests/php/test-blocks.php

```diff
@@ -330,4 +330,99 @@ public function test_jetpack_register_block_with_existing_editor_style() {
 			remove_filter( 'jetpack_is_standalone_block', '__return_false' );
 		}
 	}
+
+	/**
+	 * Test registering a block by specifying the path to its metadata file.
+	 *
+	 * @since 1.4.23
+	 *
+	 * @covers Automattic\Jetpack\Blocks::get_block_metadata_from_file
+	 */
+	public function test_jetpack_register_block_from_metadata_file() {
+		$result = Blocks::jetpack_register_block( __DIR__ . '/fixtures/block.json' );
+
+		$this->assertInstanceOf( 'WP_Block_Type', $result );
+	}
+
+	/**
+	 * Test reading metadata from a block.json file by specifying its path.
+	 *
+	 * @since 1.4.23
+	 *
+	 * @covers Automattic\Jetpack\Blocks::get_block_metadata_from_file
+	 */
+	public function test_get_block_metadata_from_file() {
+		$result = Blocks::get_block_metadata_from_file( __DIR__ . '/fixtures/block.json' );
+
+		// phpcs:ignore MediaWiki.PHPUnit.SpecificAssertions.assertIsArray -- assertIsArray not supported by all PHP versions we support.
+		$this->assertTrue( is_array( $result ) );
+		$this->assertNotEmpty( $result );
+	}
+
+	/**
+	 * Test reading metadata from a block.json file by specifying its folder.
+	 *
+	 * @since 1.4.23
+	 *
+	 * @covers Automattic\Jetpack\Blocks::get_block_metadata_from_file
+	 */
+	public function test_get_block_metadata_from_folder() {
+		$result = Blocks::get_block_metadata_from_file( __DIR__ . '/fixtures/' );
+
+		// phpcs:ignore MediaWiki.PHPUnit.SpecificAssertions.assertIsArray -- assertIsArray not supported by all PHP versions we support.
+		$this->assertTrue( is_array( $result ) );
+		$this->assertNotEmpty( $result );
+	}
+
+	/**
+	 * Test reading metadata from a file that doesn't exist.
+	 *
+	 * @since 1.4.23
+	 *
+	 * @covers Automattic\Jetpack\Blocks::get_block_metadata_from_file
+	 */
+	public function test_get_block_metadata_from_wrong_file() {
+		$result = Blocks::get_block_metadata_from_file( __DIR__ . '/fixtures/ghost-folder/block.json' );
+
+		// phpcs:ignore MediaWiki.PHPUnit.SpecificAssertions.assertIsArray -- assertIsArray not supported by all PHP versions we support.
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEmpty( $result );
+	}
+
+	/**
+	 * Test reading the name of a block from its metadata.
+	 *
+	 * @since 1.4.23
+	 *
+	 * @covers Automattic\Jetpack\Blocks::get_block_name_from_metadata
+	 */
+	public function test_get_block_name_from_metadata() {
+		$name   = 'jetpack/test-block';
+		$result = Blocks::get_block_name_from_metadata( array() );
+
+		$this->assertSame( '', $result );
+
+		$result = Blocks::get_block_name_from_metadata( array( 'name' => $name ) );
+
+		$this->assertEquals( $result, $name );
+	}
+
+	/**
+	 * Test reading the feature name of a block from its metadata.
+	 *
+	 * @since 1.4.23
```

Comment:
This will be automatically replaced by the correct version number when we next release a new version of the Blocks package (which may be 1.5.0 or something else).

```suggestion
	 * @since $$next-version$$
```

For reference:
https://github.com/Automattic/jetpack/blob/3a3c4adc1c0913bac224ddb832713bd6101fe1ab/projects/packages/README.md#L107-L116